app/controllers/proposals_controller.rb

```diff
@@ -71,4 +93,12 @@ def proposals
   def chronological_proposals
     self.proposals.order('created_at DESC')
   end
+
+  def auth_errors(exception)
```

Comment:
I'm still not sure how best to handle this. I would consider pulling this into TokenAuth, but the redirect to a different url adds a bit of complexity (knowledge about specific actions related to the different url). Thoughts?